### ISSUE TITLE
ChunkingNG: Uploading existing file re-creates fileID

### ISSUE THREAD

**Issue by guruz** [owncloud/core#26992]
The FileID returned by the final MOVE is not the same as the existing's file FileID.
Normaly FileID should not change when a file is modified.

https://github.com/owncloud/client/wiki/Etags-and-file-ids "File IDs never change for the lifetime of a file."

Note that this is only a theorical issue. The client does not have any problem with changing fileids and the sync algorithm will not be impacted by this bug.

```
01-20 19:02:26:711 528416 OCC::PropagateUploadFileNG::slotMoveJobFinished: WARN: File ID changed! "00011435ocdvdsg4rwx8" "00013624ocdvdsg4rwx8"
```

**Comment by PVince81** [comment]
@SergioBertolinSG we need integration tests for new chunking

**Comment by SergioBertolinSG** [comment]
@PVince81 there are some tests about it already: https://github.com/owncloud/core/blob/master/build/integration/features/webdav-related.feature#L374

**Comment by PVince81** [comment]
@SergioBetrolinSG please add one to cover this one case where the FileID must not change on overwrite. (also see if we have this test for dav v1 and if not, add it too)

**Comment by PVince81** [comment]
Any update ?

**Comment by SergioBertolinSG** [comment]
Added here: https://github.com/owncloud/core/pull/27508

it is failing because the fileID changes.

**Comment by PVince81** [comment]
Analysis here: https://github.com/owncloud/core/pull/27508#issuecomment-289515920.

New DAV endpoint doesn't reuse custom ObjectTree so "move" becomes a copy+delete which will cause fileid change.

**Comment by PVince81** [comment]
and fileid change is bad because it causes metadata loss, bye bye shares, favorite, tags, comments, ...

**Comment by PVince81** [comment]
It's even worse, a lot of logic from ObjectTree is now missing:
- check for invalid file names from `getNodeForPath`, so this old issue https://github.com/owncloud/core/issues/16825 now occurs on the new endpoint
- the blacklist logic "forbidden folder/file" is likely not kicking in
- move/copy

Since we can't reuse ObjectTree in the middle of the new DAV tree, we'll need to think of an alternative implementation for the checks.

One idea for these checks would be to add them in "before:method" calls in the FilesPlugin, especially because they are only relevant for files, not other nodes.

And the move stuff, reimplement in `IMoveTarget::moveInto` in the `Directory` class. No need to port all the checks there as we should do them in "beforeMethod:MOVE" then

**Comment by PVince81** [comment]
The is also special handling for chunking in `ObjectTree::getNodeForPath` but I don't think it's relevant for the new DAV endpoint.

There is also some special code related to "part" files. Not sure if relevant: https://github.com/owncloud/core/blob/v9.1.4/apps/dav/lib/Connector/Sabre/ObjectTree.php#L135

**Comment by PVince81** [comment]
Tentative fix here https://github.com/owncloud/core/pull/27508#issuecomment-290029593
Seems to work so far...

**Comment by PVince81** [comment]
Only 10.0 beta will be affected so far as only 10.0 publishes the "chunking 1.0" capability which triggers the client 2.3's use of the new DAV endpoint.

So no urgency to fix/release 9.1 for this but we need to get this sorted out before 10.0 final.

**Comment by PVince81** [comment]
Fixed in https://github.com/owncloud/core/pull/27508.
